ozaria/site/components/play/PageUnitMap/index.vue

```diff
@@ -74,23 +55,47 @@
         await this.loadUnitMapData()
       }
     },
-    async created () {
+    async mounted () {
       if (!me.showOzariaCampaign()) {
         // TODO: Remove when ready for production use
         return application.router.navigate('/', { trigger: true })
       }
-      if ((me.isStudent() && !this.courseInstanceId) || me.isTeacher()) {
-        this.notAvailable = true // showing not available text for now on the page, can update later as per requirements.
-        return
+      if ((me.isStudent() && !this.courseInstanceId)) {
+        return application.router.navigate('/students', { trigger: true })
+      } else if (me.isTeacher()) {
+        return application.router.navigate('/teachers', { trigger: true })
       }
       await this.loadUnitMapData()
+      window.addEventListener('resize', this.onWindowResize)
+      this.onWindowResize()
+      this.playAmbientSound()
+    },
+    beforeDestroy () {
+      if (this.ambientSound) {
+        createjs.Tween.get(this.ambientSound).to({ volume: 0.0 }, 1500).call(this.ambientSound.stop())
```

Comment:
The way I understand this, the static `this.ambientSound.stop()` is run right away here, and then the result of that (I can't find a return value for this call at https://createjs.com/docs/soundjs/classes/Sound.html#method_stop) is sent to to `.call()`? 

As I was reading this, I expected to see `.call(this.ambientSound.stop)`, so it would be called once the sound reached zero.

Reply:
That is a great catch, this definitely was not working as expected!! Thanks for the correction here 😄 👍